File: javascript/regimeFilter.js
/* eslint-disable no-console */
const url = new URL("https://api.cryptobob.de/v1/regime");
url.searchParams.set("timeframe", "1h");

const API_KEY = "YOUR_API_KEY";

async function main() {
  const res = await fetch(url, {
    headers: { "X-API-Key": API_KEY },
  });

  if (!res.ok) {
    const text = await res.text();
    throw new Error(`HTTP ${res.status}: ${text}`);
  }

  const data = await res.json();
  console.log(data);

  // Example decision stub:
  // if (data.is_tradeable) console.log("Tradeable regime");
  // else console.log("Skip trading");
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: python/regime_filter.py
import requests

API_KEY = "YOUR_API_KEY"

def main():
    # Adjust endpoint/params to match your API if different.
    r = requests.get(
        "https://api.cryptobob.de/v1/regime",
        headers={"X-API-Key": API_KEY},
        params={"timeframe": "1h"},
        timeout=30,
    )
    r.raise_for_status()
    data = r.json()
    print(data)

    # Example: naive decision logic (replace with your actual fields)
    # if data.get("is_tradeable"):
    #     print("Tradeable regime: OK to run strategy")
    # else:
    #     print("Untradeable regime: skip")

if __name__ == "__main__":
    main()
